app/(with-main-header)/mypage/_components/Tab.tsx

```diff
@@ -1,19 +1,33 @@
 'use client';
 
 import { TabKey } from '@/types/mypage';
-import { TAB_OPTIONS } from '@/constants/dropdown';
+import { TAB_OPTIONS_APPLICANT, TAB_OPTIONS_OWNER } from '@/constants/dropdown';
+import { useUserStore } from '@/store/user';
 
 interface TabProps {
   activeTab: TabKey;
   onTabChange: (tab: TabKey) => void;
 }
 
 const Tab = ({ activeTab, onTabChange }: TabProps) => {
+  const user = useUserStore((state) => state.user);
+
+  const getTabOptions = () => {
+    switch (user?.role) {
+      case 'APPLICANT':
+        return TAB_OPTIONS_APPLICANT;
+      case 'OWNER':
+        return TAB_OPTIONS_OWNER;
+    }
+  };
+
+  const tabOptions = getTabOptions();
```

Comment:
💬 이거 따로 getTabOptions 변수말고 tabOptions로 바로 받을 수 있을것같아서 지피티 돌려봤는데 뜻밖에 대답을 했어요..ㅋㅋㅋ
```suggestion
const tabOptions = {
  APPLICANT: TAB_OPTIONS_APPLICANT,
  OWNER: TAB_OPTIONS_OWNER,
};

const getTabOptions = () => tabOptions[user?.role];
```

Reply:
헉 훨씬 더 간단하게 짤 수 있군요 ...!!

Reply:
```javascript
const tabOptions = {
    APPLICANT: TAB_OPTIONS_APPLICANT,
    OWNER: TAB_OPTIONS_OWNER,
  };

  const currentTabOptions = tabOptions[user?.role as keyof typeof tabOptions];
```
타입정의땜에 .. 이렇게 될 것 같은데 흠 뭐가 더 좋을지 고민이 됩니다